sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -541,40 +552,31 @@ object ViewHelper {
   }
 
   /**
-   * Collect all temporary views and functions and return the identifiers separately
-   * This func traverses the unresolved plan `child`. Below are the reasons:
-   * 1) Analyzer replaces unresolved temporary views by a SubqueryAlias with the corresponding
-   * logical plan. After replacement, it is impossible to detect whether the SubqueryAlias is
-   * added/generated from a temporary view.
-   * 2) The temp functions are represented by multiple classes. Most are inaccessible from this
-   * package (e.g., HiveGenericUDF).
+   * Collect all temporary views and functions and return the identifiers separately.
    */
   private def collectTemporaryObjects(
       catalog: SessionCatalog, child: LogicalPlan): (Seq[Seq[String]], Seq[String]) = {
     def collectTempViews(child: LogicalPlan): Seq[Seq[String]] = {
       child.flatMap {
-        case UnresolvedRelation(nameParts, _, _) if catalog.isTempView(nameParts) =>
-          Seq(nameParts)
-        case w: With if !w.resolved => w.innerChildren.flatMap(collectTempViews)
-        case plan if !plan.resolved => plan.expressions.flatMap(_.flatMap {
+        case s @ SubqueryAlias(_, view: View) if view.isTempView =>
```

Comment:
is it an existing but that we missed handling `SubqueryAlias`?

Reply:
ah it's `child.flatMap`, seems we don't need to match `SubqueryAlias` here?

Reply:
BTW does temp view come with `SubqueryAlias`? What's the qualifier there?

Reply:
> ah it's `child.flatMap`, seems we don't need to match `SubqueryAlias` here?

Ah, yes. `SubqueryAlias` is not needed any more as `View` always has `CatalogTable`.
 
> BTW does temp view come with `SubqueryAlias`? What's the qualifier there?

Yes, temp views come with `SubqueryAlias` if used with `LookupTempView`:
https://github.com/apache/spark/blob/7f27d33a3c538da6754a6c011b29aa7eb0dafe2c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L884-L889

The qualifier is empty for temp view, and global temp database name for global temp view.

Reply:
BTW, this is not complete until we migrate `ALTER VIEW AS` and `CREATE TEMP VIEW USING` since they may store analyzed plan directly. Do you want to me update these commands in this PR?

Reply:
Let's do it in followups.